Source/Dafny/Compilers/Compiler-js.cs

```diff
@@ -17,13 +17,18 @@
 using System.Threading.Tasks;
 using Bpl = Microsoft.Boogie;
 
-namespace Microsoft.Dafny {
-  public class JavaScriptCompiler : Compiler {
-    public JavaScriptCompiler(ErrorReporter reporter)
-    : base(reporter) {
-    }
+namespace Microsoft.Dafny.Compilers.Js {
+  public class JavaScriptCompiler : SinglePassCompiler {
+    public override IReadOnlySet<string> SupportedExtensions => new HashSet<string> { ".js" };
 
     public override string TargetLanguage => "JavaScript";
+    public override string TargetExtension => "js";
+
+    public override bool SupportsInMemoryCompilation => true;
+    public override bool TextualTargetIsExecutable => true;
+
+    public override IReadOnlySet<string> SupportedNativeTypes =>
+      new HashSet<string>(base.SupportedNativeTypes.Union(new List<string> { "number" }));
```

Comment:
In the previous huge switch statement, the JavaScript supported native types was only "number". Why are you unioning it with the base.SupportedNativeTypes? Can you confirm it is safe?